### PR TITLE
Add download_iau2000_file()

### PR DESCRIPTION
**Pull request by EugeneDu-GA**
Add download_iau2000_file() and other improvements

### CODE REVIEW

File: gnssanalysis/gn_download.py [hunk 1]
```diff
@@ -830,12 +831,15 @@ def download_product_from_cddis(
     logging.debug(
         f"Generated filename: {product_filename}, with GPS Date: {gps_date.gpswkD} and reference: {reference_start}"
     )
+
+    ensure_folders([download_dir])
+    download_filepaths = []
```
Comment (ronaldmaj) [inline gnssanalysis/gn_download.py:836]: This is not returned at the end (if that's what we want to change the behaviour to)

File: gnssanalysis/gn_download.py [hunk 2]
```diff
@@ -877,17 +881,68 @@ def download_product_from_cddis(
...
+                            note_filetype=file_ext,
+                        )
                     )
                 count += 1
                 remain = end_epoch - reference_start
 
 
+def download_iau2000_file(download_dir: _Path, start_epoch: _datetime, if_file_present: str = "prompt_user"):
+    """
+    Download relevant IAU2000 file from CDDIS or IERS based on start_epoch of data
```
Comment (ronaldmaj) [inline gnssanalysis/gn_download.py:899]: docstring will need to be expanded out here

File: gnssanalysis/gn_download.py [hunk 3]
```diff
@@ -503,27 +503,27 @@ def attempt_ftps_download(
 
 
 def attempt_url_download(
-    download_dir: _Path, url: str, filename: str = None, type_of_file: str = None, if_file_present: str = "prompt_user"
+    download_dir: _Path, url: str, download_filename: str = None, type_of_file: str = None, if_file_present: str = "prompt_user"
```
Comment (ronaldmaj) [inline gnssanalysis/gn_download.py:506]: If we are changing `download_filename` here, we'll need to make sure all functions that call this, use the new argument name

**Review by ronaldmaj — CHANGES_REQUESTED** [review]
This is looking good. The changes to the `attempt_download` function look ok, but we'll need to make sure that all functions that rely on it are ok

File: gnssanalysis/gn_download.py (hunk 3 above)
Comment (EugeneDu-GA) [reply gnssanalysis/gn_download.py:506]: Changed back to `filename` as it might affect other external scripts including `auto_download_PPP.py`

**Review by ronaldmaj — CHANGES_REQUESTED** [review]
Looking really good, thanks for addressing my previous comments. Just a formatting issue I found in that the line length seems not to be set at 120? After that I think this can go in

**Review by ronaldmaj — APPROVED** [review]
Cool, thanks for fixing up the formatting. I think this is good to go in now

**Pull request merged**